xdsl/transforms/convert_pdl_to_pdl_interp/predicate.py

```diff
@@ -0,0 +1,464 @@
+"""
+This file implements some of the core data structures used in the pdl-to-pdl-interp conversion.
+The pdl-to-pdl-interp conversion works by translating declarative [pdl patterns](../../../dialects/pdl) into a set of
```

Comment:
The relative link doesn't seem to work, can you please try this?
```suggestion
The pdl-to-pdl-interp conversion works by translating declarative [pdl patterns](xdsl.dialects.pdl) into a set of
```

Reply:
I'd tried but it didn't work and force pushed over it. `../../../dialects/pdl` works but I'll try `xdsl.dialects.pdl` again in a proper commit.

Reply:
Kk I'll have more of a look, I thought this kind of thing should work.